rts/Lua/LuaOpenGL.cpp

```diff
@@ -5667,5 +5673,74 @@ int LuaOpenGL::GetMapRendering(lua_State* L)
 	return 0;
 }
 
+/**
+ * @function gl.ObjectLabel labels an object for use with debugging tools
+ * @param objectTypeIdentifier GLenum Specifies the type of object being labeled.
+ * @param objectID GLuint Specifies the name or ID of the object to label.
+ * @param label string A string containing the label to be assigned to the object.
+ * @treturn nil
+ */
+int LuaOpenGL::ObjectLabel(lua_State* L) {
+	const auto identifier = static_cast<GLenum>(luaL_checkinteger(L, 1));
+
+	switch (identifier) {
+	case GL_BUFFER: [[fallthrough]];
+	case GL_SHADER: [[fallthrough]];
+	case GL_PROGRAM: [[fallthrough]];
+	case GL_VERTEX_ARRAY: [[fallthrough]];
+	case GL_QUERY: [[fallthrough]];
+	case GL_PROGRAM_PIPELINE: [[fallthrough]];
+	case GL_TRANSFORM_FEEDBACK: [[fallthrough]];
+	case GL_TEXTURE: [[fallthrough]];
+	case GL_RENDERBUFFER: [[fallthrough]];
+	case GL_FRAMEBUFFER:
+		break;
+	default: {  // something else
+		LOG_L(L_ERROR, "gl.%s: invalid identifier (%u)", __func__, identifier);
+		return 0;
+	}
+	}
+
+	const auto objectID = static_cast<GLuint>(luaL_checkinteger(L, 2));
+	const auto* label = luaL_checkstring(L, 3);
+	glObjectLabel(identifier, objectID, -1, label);
+
+	return 0;
+}
+
+// https://registry.khronos.org/OpenGL-Refpages/gl4/html/glPushDebugGroup.xhtml
+/**
+ * @function gl.PushDebugGroup pushes a debug marker for nVidia nSight 2024.04, does not seem to work when FBO's are raw bound
+ * @param id GLuint A numeric identifier for the group.
+ * @param message string A human-readable string describing the debug group.
+ * @param source boolean true for GL_DEBUG_SOURCE_APPLICATION, false for GL_DEBUG_SOURCE_THIRD_PARTY. default false
```

Comment:
Is gameside Lua code ever actually "third party"? Doesn't that mean libraries like SDL or GLEW/GLAD or something along those lines?

Reply:
Sounds like it's always `DEBUG_SOURCE_APPLICATION` then but maybe @Beherith would know how he intended it to be used. Not a blocker either way though.

Reply:
Indeed, it should probably be default true. I don't yet have any intent of fully third party, but didn't want to leave the option closed, cause then y'all would complain about that bit :P

Reply:
Changed default to `DEBUG_SOURCE_APPLICATION` 0dec6ee92e9383d24757d64636ac3a839c3c7f13

Ideally this would let you specify the tag directly (e.g. pass a `GL.DEBUG_SOURCE_APPLICATION` constant instead of a bool, because that's much better for both the range of possible values and self-documentation) but I don't expect anybody to really change it from the default so I don't mind it staying a bool for now.